src/server/components/dashboard/controllers/listExportController.ts

```diff
@@ -0,0 +1,91 @@
+import type { NextFunction, Request, Response } from "express";
+import type { ListItem } from "server/models/types";
+import type { ListItemJsonData } from "server/models/listItem/providers/deserialisers/types";
+
+import { prisma } from "server/models/db/prisma-client";
+import { json2csv } from "json-2-csv";
+import { Readable } from "node:stream";
+import { startCase } from "lodash";
+import { format } from "date-fns";
+import { logger } from "server/services/logger";
+import { AuditEvent } from "@prisma/client";
+
+type ListItemWithJsonData = ListItem & { jsonData: ListItemJsonData };
+
+export async function listExportController(req: Request, res: Response, next: NextFunction) {
+  try {
+    const result = await prisma.list.findUniqueOrThrow({
+      where: {
+        id: Number(req.params.listId),
+      },
+      include: {
+        items: true,
+        country: true,
+      },
+    });
+
+    const formattedItems = (result.items as ListItemWithJsonData[]).map(formatForCSV);
+
+    const output = await json2csv(formattedItems, {
+      emptyFieldValue: "none provided",
+      expandNestedObjects: false,
+      excludeKeys: [
+        "updatedAt",
+        "addressId",
+        "listId",
+        "metadata",
+        "declaration",
+        "updatedJsonData",
+        "addressCountry",
+        "type",
+        "country",
+      ],
+      preventCsvInjection: true,
+    });
+
+    const currentDate = new Date();
+    const formattedDate = format(currentDate, "yyyy-MM-dd");
+    const fileName = `${result?.country.name}-${startCase(result?.type)}-${formattedDate}.csv`;
```

Comment:
is `result` still undefined? do we need `?.` anymore?

Reply:
Yep you're right, the `?` isn't needed